gbpservice/neutron/nsf/core/main.py

```diff
@@ -0,0 +1,582 @@
+            ev = self._get_ev()
+            if ev:
+                ev.worker = self
+                eh = self._eh.get(ev)
+                if not ev.poll_event:
+                    self._tpool.dispatch(eh.handle_event, ev)
+                else:
+                    self._tpool.dispatch(eh.handle_poll_event, ev)
+            time.sleep(0)  # Yield the CPU
+    '''
+
+
+class EventHandlers(object):
+
+    def __init__(self):
+        self._ehs = {}
+
+    def register(self, ev):
+        if ev.id in self._ehs.keys():
+            self._ehs[ev.id].extend([ev])
+        else:
+            self._ehs[ev.id] = [ev]
+
+    def get(self, ev):
+        for id, eh in self._ehs.iteritems():
+            if id == ev.id:
+                return eh[0].handler
+        return None
+
+
+class ServiceController(object):
+
+    def __init__(self, conf, modules):
+        self._conf = conf
+        self.modules = modules
+        self._lock = Lock()
+        self._serializer = Serializer(self)
+
+    def lock(self):
+        self._lock.acquire()
+
+    def unlock(self):
+        self._lock.release()
+
+    def event_done(self, ev):
+        mapp = self._serializer.copy()
+        mapp = mapp[ev.worker_attached]
+        if ev.binding_key not in mapp:
+            return
+
+        qu = mapp[ev.binding_key]['queue']
+        for elem in qu:
+            if elem.key == ev.key:
+                self._serializer.remove(elem)
+                break
+        self._serializer.deserialize(ev)
+
+    def serialize(self, ev):
+        if not ev.serialize:
+            return ev
+        if not self._serializer.serialize(ev):
+            return ev
+        return None
+
+    def serialize_get(self):
+        smap = self._serializer.copy()
+        for mapp in smap.values():
+            for val in mapp.values():
+                if val['in_use']:
+                    continue
+                else:
+                    if val['queue'] == []:
+                        continue
+                    return val['queue'][0]
+        return None
+
+    def workers_init(self):
+        wc = 2 * (multiprocessing.cpu_count())
+        if cfg.CONF.workers != wc:
+            wc = cfg.CONF.workers
+            LOG.debug("Creating configured #of workers:%d" % (wc))
+
+        workers = [tuple() for w in range(0, wc)]
+
+        for w in range(0, wc):
+            # qu = LookAheadQueue()
+            qu = Queue()
+            evworker = EventHandler(self, qu, self.ehs)
+            proc = Process(target=evworker.run, args=(qu,))
+            proc.daemon = True
+            workers[w] = workers[w] + (proc, qu, evworker)
+        return workers
+
+    def poll_init(self):
+        # qu = LookAheadQueue()
+        qu = Queue()
+        ph = PollHandler(self, qu, self.ehs)
+        return ph
+
+    def modules_init(self, modules):
+        for module in modules:
+            try:
+                module.module_init(self, self._conf)
+            except AttributeError as s:
```

Comment:
At that line we are trying to access an attribute of a class -- so only possibility is if that attribute is not present. what other exceptions do you expect ?